exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/ContextFunctions.java

```diff
@@ -64,17 +65,45 @@ public void eval() {
     @Inject DrillBuf buffer;
     @Workspace int currentSchemaBytesLength;
 
+    @Override
     public void setup() {
       final byte[] currentSchemaBytes = contextInfo.getCurrentDefaultSchema().getBytes();
       buffer = buffer.reallocIfNeeded(currentSchemaBytes.length);
       currentSchemaBytesLength= currentSchemaBytes.length;
       buffer.setBytes(0, currentSchemaBytes);
     }
 
+    @Override
     public void eval() {
       out.start = 0;
       out.end = currentSchemaBytesLength;
       out.buffer = buffer;
     }
   }
+
+  /**
+   * Implement "session_id" function. Returns the unique id of the current session.
+   */
+  @FunctionTemplate(name = "session_id", scope = FunctionTemplate.FunctionScope.SIMPLE, isNiladic = true)
```

Comment:
- Functions like current_time/current_date already function like niladic as Calcite recognizes them so. 

- IsNiladic flag can be set by any UDF. Not restricted only to session_I'd

- If I understand right, currently there is no categorization of UDFs in Drill. If we want to restrict isNiladic flag only to a particular category of UDFs, then categorization of UDFs need to be designed n implemented.